### PR TITLE
Fix frontend robots.txt

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Fixes a regression introduced in #4077 

## Description

This PR returns to using a sever middleware for robots.txt in the Openverse frontend. It also adds robots.txt to the frontend gitignore to help prevent us from accidentally adding a static robots.txt file in the future.

It also adds Crawl-delay to slow down bing and other search engines when they crawl the pages we _do_ allow them to crawl.

## Testing Instructions

Run the frontend locally with `just frontend/run dev` and verify that you see the correct file contents as defined in [/frontend/src/server-middleware/robots.js](https://github.com/WordPress/openverse/blob/fix-frontend-robots/frontend/src/server-middleware/robots.js#L65)

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/server-middleware/robots.js [hunk 1]
```diff
@@ -10,13 +50,17 @@ export default function robots(_, res) {
...
 Disallow: /search/image/
 Disallow: /search/
 Disallow: /image/
 Disallow: /audio/
+
+crawl-delay:
+${AI_ROBOTS_CONTENT}
       `
-      : `# Block crawlers from the staging site
+      : `# Block everyone from the staging site
```
Comment (sarayourfriend) [inline frontend/src/server-middleware/robots.js:63]: Ah! I suppose we should have a similar one for the staging API docs, then?

File: frontend/.gitignore [hunk 2]
```diff
@@ -27,3 +27,7 @@ src/locales/scripts/wp-locales.json
 /.npmrc
 /.pnpmfile.cjs
 /pnpm-lock.yaml
+
+# To prevent accidentally adding a hardcoded robots.txt, see
+# /src/server-middleware/robots.js for the robots.txt file.
+src/static/robots.txt
```
Comment (sarayourfriend) [inline frontend/.gitignore:33]: :facepalm: Sorry I missed this! I thought I had looked for an existing robots thing in the frontend, but I must only have looked in nuxt.config and not seen anything there. Not sure why I didn't just _search_ for it :sweat_smile:

**Review by sarayourfriend — APPROVED** [review]
Good work figuring out what changed with the traffic!

File: frontend/src/server-middleware/robots.js [hunk 3]
```diff
@@ -1,5 +1,45 @@
 const { LOCAL, PRODUCTION } = require("../constants/deploy-env")
 
+const AI_ROBOTS_CONTENT = `
```
Comment (obulat) [inline frontend/src/server-middleware/robots.js:3]: Would it be easier to maintain  a list of user-agent names, and create the block string using it:
```
uaList.map(ua => `User-agent: ${ua}\nDisallow: /\n`).join("\n")
```

File: frontend/src/server-middleware/robots.js [hunk 4]
```diff
@@ -10,13 +50,17 @@ export default function robots(_, res) {
     deployEnv === PRODUCTION
       ? `# Block search result pages
```
Comment (obulat) [inline frontend/src/server-middleware/robots.js:51]:
```suggestion
      ? `# Block search result pages and single result pages
```

**Review by obulat — APPROVED** [review]
LGTM! I left some non-blocking comments and questions inline.

Note on the testing instructions: you need to run `DEPLOYMENT_ENV=production just frontend/run dev:only` to see the production values, otherwise you'll see the "block everyone" robots.txt :)

File: frontend/src/server-middleware/robots.js [hunk 5]
```diff
@@ -10,13 +50,17 @@ export default function robots(_, res) {
...
 User-agent: *
+Crawl-delay: 10
 Disallow: /search/audio/
 Disallow: /search/image/
 Disallow: /search/
 Disallow: /image/
 Disallow: /audio/
+
+crawl-delay:
```
[the remark this answers is not in the document]
Comment (zackkrida) [reply frontend/src/server-middleware/robots.js]: Typo!

**Pull request merged**